be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -137,157 +180,269 @@ object DbActor extends AskPatternConstants {
+  // FIXME: find way to keep count of channel nb. Idea: store key "channelCount" with value int.getBytes to db and increment it with each channel created
+  sealed case class DbActor(mediatorRef: ActorRef, db: DB, DATABASE_FOLDER: String) extends Actor with ActorLogging {
 
     override def preStart(): Unit = {
-      log.info(
-        s"Actor $self (db) was initialised with a total of ${initialChannelsMap.size} recovered channels"
-      )
-      if (initialChannelsMap.size > DATABASE_MAX_CHANNELS) {
-        log.warning(
-          s"Actor $self (db) has surpassed a large number of active lao channels (${initialChannelsMap.size} > $DATABASE_MAX_CHANNELS)"
-        )
-      }
-
+      //FIXME: check if this should be kept
       super.preStart()
     }
 
-    private def createDatabase(channel: Channel): DB = {
-      val options: Options = new Options()
-      options.createIfMissing(true)
-
-      val channelName: String = s"$DATABASE_FOLDER$channel"
-      val channelDb = factory.open(new File(channelName), options)
-
-      channelNamesDb.put(channelName.getBytes, "".getBytes)
-      channelsMap += (channel -> channelDb)
-
-      channelDb
-    }
-
+    // only for the messages
     private def write(channel: Channel, message: Message): DbActorMessage = {
-      val channelDb: DB = channelsMap.get(channel) match {
-        case Some(channelDb) => channelDb
-        case _ => createDatabase(channel)
+      
+      //encapsulate this inside function
+      Try(db.get(channel.toString.getBytes)) match {
+        case Success(bytes) if bytes != null =>
+          log.info(s"Channel '$channel' already exists.")
+        case _ =>
+          //FIXME: handle type of message (is it actually a problem or is the MessageExample coded in a weird way?)
+          //val objectType = message.decodedData.getOrElse(null)._object //or some other way, then match using match case and establish channel type
+          val objectType = ObjectType.LAO
+          val actualObjectType = if (objectType == ObjectType.ELECTION || objectType == ObjectType.CHIRP){
+            objectType
+          } else {
+            ObjectType.LAO
+          }
+          createChannel(channel, actualObjectType) match {
+            case DbActorAck() =>
+            case _ =>
+              log.info(s"Actor $self (db) encountered a problem while creating channel '$channel'")
+              DbActorNAck(ErrorCodes.SERVER_ERROR.id, s"Error when creating channel '$channel'")
+          }
       }
-
+      
       val messageId: Hash = message.message_id
-      Try(channelDb.put(messageId.getBytes, message.toJsonString.getBytes)) match {
-        case Success(_) =>
-          log.info(s"Actor $self (db) wrote message_id '$messageId' on channel '$channel'")
-          DbActorWriteAck
+      Try(db.get(channel.toString.getBytes)) match {
+        case Success(bytes) if bytes != null => {
+          val key: String = channel + "/" + messageId.toString
```

Comment:
It would be nice to have some kind of help methods to compute the key from its parameters. 
So it will convey more abstraction and less confusion (and thus errors) for future developers.

Reply:
Indeed, I totally agree with you